main.tf

```diff
@@ -1 +1,346 @@
+    command = "${join(" ", local.s3_rm)}"
+    when    = "destroy"
+  }
+
+  triggers = {
+    s3_sync_salt      = "${join(" ", local.s3_sync_salt)}"
+    s3_sync_appscript = "${join(" ", local.s3_sync_appscript)}"
+  }
+}
+
+# Manage domain record
+resource "aws_route53_record" "www" {
+  count   = "${ var.route53_enabled ? 1 : 0 }"
+  zone_id = "${ var.zone_id }"
+  name    = "${ local.plain_domain }"
+  type    = "A"
+
+  alias {
+    name                   = "${ aws_lb.this.dns_name }"
+    zone_id                = "${ aws_lb.this.zone_id }"
+    evaluate_target_health = false
+  }
+}
+
+# Manage load balancer
+resource "aws_lb" "this" {
+  name            = "${var.name}-lb-${var.environment}"
+  internal        = "false"
+  security_groups = ["${aws_security_group.lb.id}"]
+  subnets         = ["${var.lb_subnet_ids}"]
+
+  # access_logs {
+  #   enabled = true
+  #   bucket  = "${module.bucket.bucket_name}"
+  #   prefix  = "logs/lb_access_logs"
+  # }
+
+  tags = "${ merge(map("Name", "${var.name}-lb"), local.tags ) }"
+}
+
+# redirects HTTP requests to HTTPS
+resource "aws_lb_listener" "http" {
+  load_balancer_arn = "${ aws_lb.this.arn }"
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type = "redirect"
+
+    redirect {
+      port        = "443"
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+resource "aws_lb_listener" "https" {
+  load_balancer_arn = "${ aws_lb.this.arn }"
+  port              = "443"
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-FS-2018-06"
+  certificate_arn   = "${var.lb_certificate_arn}"
+
+  default_action {
+    target_group_arn = "${ aws_lb_target_group.this.arn }"
+    type             = "forward"
+  }
+}
+
+resource "aws_lb_target_group" "this" {
+  name     = "${var.name}-${var.environment}"
+  port     = "8200"
+  protocol = "HTTP"
+  vpc_id   = "${ local.vpc_id }"
+
+  deregistration_delay = "10"
+
+  # /sys/health will return 200 only if the vault instance
+  # is the leader. Meaning there will only ever be one healthy
+  # instance, but a failure will cause a new instance to
+  # be healthy automatically. This healthceck path prevents
+  # unnecessary redirect loops by not sending traffic to
+  # followers, which always just route traffic to the master
+  health_check {
+    path                = "/v1/sys/health?standbyok=true"
+    port                = "8200"
+    interval            = "5"
+    timeout             = "3"
+    healthy_threshold   = "2"
+    unhealthy_threshold = "2"
+  }
+
+  tags = "${ merge(
+    map("Name","${ var.name }-tg"),
+    local.tags ) }"
+}
+
+#Security Groups
+resource "aws_security_group" "lb" {
+  name        = "${var.name}-${var.environment}"
+  description = "Rules required for operation of ${var.name}"
+  vpc_id      = "${local.vpc_id}"
+
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
```

Comment:
recommend a variable, `var.ingress_cidr_blocks` that defaults to `["0.0.0.0/0"]`, so the user can control this